src/shared/components/post/post-listing.tsx

```diff
@@ -315,26 +319,12 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     );
   }
 
-  get imageSrc(): string | undefined {
-    const post = this.postView.post;
-    const url = post.url;
-    const thumbnail = post.thumbnail_url;
-
-    if (thumbnail) {
-      return thumbnail;
```

Comment:
The culprit was here. It was early returning thumbnail, even for full size images. I got rid of this whole function to be more explicit, and moved the logic into the specific functions.